plugin/storage/grpc/shared/interface.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2019 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package shared
+
+import (
+	"context"
+
+	"github.com/hashicorp/go-plugin"
+	"google.golang.org/grpc"
+
+	"github.com/jaegertracing/jaeger/proto-gen/storage_v1"
+	"github.com/jaegertracing/jaeger/storage/dependencystore"
+	"github.com/jaegertracing/jaeger/storage/spanstore"
+)
+
+// StoragePluginIdentifier is the identifier that is shared by plugin and host.
+const StoragePluginIdentifier = "storage_plugin"
+
+// Handshake is a common handshake that is shared by plugin and host.
+var Handshake = plugin.HandshakeConfig{
+	MagicCookieKey:   "STORAGE_PLUGIN",
+	MagicCookieValue: "jaeger",
+}
+
+// PluginMap is the map of plugins we can dispense.
+var PluginMap = map[string]plugin.Plugin{
+	StoragePluginIdentifier: &StorageGRPCPlugin{},
+}
+
+// StoragePlugin is the interface we're exposing as a plugin.
+type StoragePlugin interface {
+	SpanReader() spanstore.Reader
+	SpanWriter() spanstore.Writer
+	DependencyReader() dependencystore.Reader
+}
+
+// StorageGRPCPlugin is the implementation of plugin.GRPCPlugin so we can serve/consume this.
+type StorageGRPCPlugin struct {
+	plugin.Plugin
+	// Concrete implementation, written in Go. This is only used for plugins
+	// that are written in Go.
+	Impl StoragePlugin
+}
+
+// GRPCServer is used by go-plugin to create a grpc plugin server
+func (p *StorageGRPCPlugin) GRPCServer(broker *plugin.GRPCBroker, s *grpc.Server) error {
```

Comment:
We could probably add simple tests exercising GRPCServer/GRPCClient methods. I was actually thinking that it would be great to have tests that actually spin up the server and use the client to talk to it. I suppose this is what the integration test is doing, so we should find a way to use it to enhance coverage.


Note to self:
```shell
STORAGE=grpc-plugin go test -v -coverpkg ./plugin/storage/grpc/shared -coverprofile coverage.out ./plugin/storage/integration
```